src/main/java/edu/cornell/kfs/rass/batch/service/impl/RassReportServiceImpl.java

```diff
@@ -0,0 +1,302 @@
+import org.kuali.kfs.krad.bo.PersistableBusinessObject;
+import org.kuali.kfs.module.cg.businessobject.Agency;
+import org.kuali.kfs.module.cg.businessobject.Award;
+import org.kuali.kfs.module.cg.businessobject.Proposal;
+import org.kuali.kfs.sys.mail.BodyMailMessage;
+import org.kuali.kfs.sys.service.EmailService;
+
+import edu.cornell.kfs.rass.RassConstants;
+import edu.cornell.kfs.rass.RassConstants.RassObjectUpdateResultCode;
+import edu.cornell.kfs.rass.RassParameterConstants;
+import edu.cornell.kfs.rass.batch.RassBatchJobReport;
+import edu.cornell.kfs.rass.batch.RassBusinessObjectUpdateResult;
+import edu.cornell.kfs.rass.batch.RassBusinessObjectUpdateResultGrouping;
+import edu.cornell.kfs.rass.batch.RassStep;
+import edu.cornell.kfs.rass.batch.RassXmlFileParseResult;
+import edu.cornell.kfs.rass.batch.RassXmlProcessingResults;
+import edu.cornell.kfs.rass.batch.service.RassReportService;
+import edu.cornell.kfs.sys.service.ReportWriterService;
+import edu.cornell.kfs.sys.util.LoadFileUtils;
+
+public class RassReportServiceImpl implements RassReportService {
+    private static final Logger LOG = LogManager.getLogger(RassReportServiceImpl.class);
+
+    protected EmailService emailService;
+    protected ParameterService parameterService;
+    protected ReportWriterService reportWriterService;
+
+    @Override
+    public void sendReportEmail() {
+        BodyMailMessage message = new BodyMailMessage();
+        String fromAddress = getFromAddress();
+        Collection<String> toAddresses = getToAddressses();
+        message.setFromAddress(fromAddress);
+        String subject = reportWriterService.getTitle();
+        message.setSubject(subject);
+        message.getToAddresses().addAll(toAddresses);
+        String body = getFileContents(reportWriterService.getReportFile().getAbsolutePath());
+        message.setMessage(body);
+
+        boolean htmlMessage = false;
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("sendEmail, from address: " + fromAddress + "  to address: " + toAddresses.toString());
+            LOG.debug("sendEmail, the email subject: " + subject);
+            LOG.debug("sendEmail, the email budy: " + body);
+        }
+        try {
+            emailService.sendMessage(message, htmlMessage);
+        } catch (Exception e) {
+            LOG.error("sendEmail, the email could not be sent", e);
+        }
+    }
+
+    private Collection<String> getToAddressses() {
+        return parameterService.getParameterValuesAsString(RassStep.class, RassParameterConstants.TO_REPORT_EMAIL_ADDRESSES);
+    }
+
+    private String getFromAddress() {
+        return parameterService.getParameterValueAsString(RassStep.class, RassParameterConstants.FROM_REPORT_EMAIL_ADDRESS);
+    }
+
+    public String getFileContents(String fileName) {
+        try {
+            byte[] fileByteArray = LoadFileUtils.safelyLoadFileBytes(fileName);
+            String formattedString = new String(fileByteArray);
```

Comment:
Should you specify an explicit encoding here, or is the platform-default encoding sufficient?

Reply:
Good question, encoding should we use? Does the code that writes the report also need to be updated to use a specific encoding?

I guess since the report writing and the reading are done on same platform the platform-default should be fine?

Reply:
Yes, I agree that the platform default should be okay, since the report writer appears to be using a PrintStream configured with the default platform encoding.